askama/src/filters/default.rs

```diff
@@ -0,0 +1,403 @@
+        Self: 'a;
+
+    type Error = Infallible;
+
+    #[inline]
+    fn as_filtered(&self) -> Result<Option<&str>, Infallible> {
+        match self.is_empty() {
+            false => Ok(Some(self)),
+            true => Ok(None),
+        }
+    }
+}
+
+/// A [`String`][alloc::string::String] has a value if it is not empty.
+#[cfg(feature = "alloc")]
+impl DefaultFilterable for alloc::string::String {
+    type Filtered<'a>
+        = &'a str
+    where
+        Self: 'a;
+
+    type Error = Infallible;
+
+    #[inline]
+    fn as_filtered(&self) -> Result<Option<&str>, Infallible> {
+        self.as_str().as_filtered()
+    }
+}
+
+/// A [`Cow`][alloc::borrow::Cow] has a value if it's borrowed data has a value.
+#[cfg(feature = "alloc")]
+impl<T: DefaultFilterable + alloc::borrow::ToOwned + ?Sized> DefaultFilterable
+    for alloc::borrow::Cow<'_, T>
+{
+    type Filtered<'a>
+        = T::Filtered<'a>
+    where
+        Self: 'a;
+
+    type Error = T::Error;
+
+    #[inline]
+    fn as_filtered(&self) -> Result<Option<Self::Filtered<'_>>, Self::Error> {
+        self.as_ref().as_filtered()
+    }
+}
+
+/// A [`Wrapping`] integer has a value if it is not `0`.
+impl<T: DefaultFilterable> DefaultFilterable for Wrapping<T> {
+    type Filtered<'a>
+        = T::Filtered<'a>
+    where
+        Self: 'a;
+
+    type Error = T::Error;
+
+    #[inline]
+    fn as_filtered(&self) -> Result<Option<Self::Filtered<'_>>, Self::Error> {
+        self.0.as_filtered()
+    }
+}
+
+/// A [`Saturating`] integer has a value if it is not `0`.
+impl<T: DefaultFilterable> DefaultFilterable for Saturating<T> {
+    type Filtered<'a>
+        = T::Filtered<'a>
+    where
+        Self: 'a;
+
+    type Error = T::Error;
+
+    #[inline]
+    fn as_filtered(&self) -> Result<Option<Self::Filtered<'_>>, Self::Error> {
+        self.0.as_filtered()
+    }
+}
+
+macro_rules! impl_for_int {
+    ($($ty:ty)*) => { $(
+        #[doc = concat!("A [`", stringify!($ty), "`] has a value if it is not `0`.")]
+        impl DefaultFilterable for $ty {
+            type Filtered<'a> = $ty;
+            type Error = Infallible;
+
+            #[inline]
+            fn as_filtered(&self) -> Result<Option<$ty>, Infallible> {
+                match *self {
+                    0 => Ok(None),
+                    value => Ok(Some(value)),
+                }
+            }
+        }
+    )* };
+}
+
+impl_for_int!(
+    u8 u16 u32 u64 u128 usize
+    i8 i16 i32 i64 i128 isize
+);
+
+macro_rules! impl_for_non_zero {
+    ($($name:ident : $ty:ty)*) => { $(
+        #[doc = concat!("A [`", stringify!($name), "`][num::", stringify!($name),"] always has a value.")]
+        impl DefaultFilterable for num::$name {
+            type Filtered<'a> = $ty;
+            type Error = Infallible;
+
+            #[inline]
+            fn as_filtered(&self) -> Result<Option<$ty>, Infallible> {
+                Ok(Some(self.get()))
+            }
+        }
+    )* };
+}
+
+impl_for_non_zero!(
```

Comment:
Since we need to provide both the non-zero ty and the integer ty, why not merge this macro with `impl_for_int`? Like that: less code and one less macro definition/expansion.